x-pack/elastic-agent/pkg/agent/cmd/run.go

```diff
@@ -296,3 +314,52 @@ func setupMetrics(agentInfo *info.AgentInfo, logger *logger.Logger, operatingSys
 func isProcessStatsEnabled(cfg *monitoringCfg.MonitoringHTTPConfig) bool {
 	return cfg != nil && cfg.Enabled
 }
+
+func tryDelayEnroll(ctx context.Context, logger *logger.Logger, cfg *configuration.Configuration, override cfgOverrider) (*configuration.Configuration, error) {
+	enrollPath := paths.AgentEnrollFile()
+	if _, err := os.Stat(enrollPath); os.IsNotExist(err) {
```

Comment:
err other than FileNotExist is ignored

Reply:
True. Updated it to check check for `err != nil`, so any error will cause it to skip the delayed enroll.